### PR TITLE
Fetch reports for the list of authors of replies in a note

### PR DESCRIPTION
**Pull request by martindsq**
- Reports for replies were being fetched, so it wasn't necessary to work on that (besides testing it).
- Now we fetch reports for authors of those replies
- Found a bug in NoteWarningController where the logical operator that checked for reports of a note or the author of that note didn't consider the reports for the author.
- Prevent reports of authors from being deleted during cleanupTask

#728
#638

### CODE REVIEW

File: Nos/Models/Persistence.swift [hunk 1]
```diff
@@ -237,8 +237,14 @@ class PersistenceController {
                 let oldEventClause = "(receivedAt <= %@ OR receivedAt == nil)"
                 let notOwnEventClause = "(author.hexadecimalPublicKey != %@)"
                 let readStoryClause = "(isRead = 1 AND receivedAt > %@)"
+                let userReportClause = "(kind == \(EventKind.report.rawValue) AND " +
+                    "authorReferences.@count > 0 AND authorReferences.@count > 0 AND eventReferences.@count == 0)"
```
Comment (martindsq) [inline Nos/Models/Persistence.swift:241]: The `eventReferences.@count == 0` part is becuase some event report messages also include a p tag and I assume those cases only mean to report the content not the whole author

File: Nos/Controller/NoteWarningController.swift [hunk 2]
```diff
@@ -30,7 +30,7 @@ extension Publisher {
     var showWarning: Bool {
         if userHidWarning {
             return false
-        } else if showReportWarnings && !(noteReports.isEmpty || !authorReports.isEmpty) {
+        } else if showReportWarnings && (!noteReports.isEmpty || !authorReports.isEmpty) {
```
Comment (mplorentz) [inline Nos/Controller/NoteWarningController.swift:33]: oops that's my bug

File: Nos/Models/Persistence.swift (hunk 1 above)
Comment (mplorentz) [reply Nos/Models/Persistence.swift:241]: It looks like the `authorReferences.@count > 0` check is in there twice? We can remove one of them right?
```suggestion
                    "authorReferences.@count > 0 AND eventReferences.@count == 0)"
```

File: Nos/Views/RepliesView.swift [hunk 3]
```diff
@@ -90,11 +90,13 @@ struct RepliesView: View {
             let subID = await relayService.openSubscription(with: filter)
             subscriptionIDs.append(subID)
             
-            // download reports for this user
+            // download reports for this user and the replies' authors
             guard let authorKey = note.author?.hexadecimalPublicKey else {
                 return
             }
-            let reportFilter = Filter(kinds: [.report], pTags: [authorKey])
+            let pTags = [authorKey] + replies.compactMap { $0.author?.hexadecimalPublicKey }
```
Comment (mplorentz) [inline Nos/Views/RepliesView.swift:97]: Not sure how relays will behave if there are duplicate author keys in this array. Can we do a .removeDuplicates on it or something?

File: Nos/Views/RepliesView.swift [hunk 4]
```diff
@@ -90,11 +90,13 @@ struct RepliesView: View {
...
+            let reportFilter = Filter(kinds: [.report], pTags: pTags)
+            // let reportFilter = Filter(kinds: [.report], pTags: [authorKey])
```
Comment (mplorentz) [inline Nos/Views/RepliesView.swift:99]:
```suggestion
```

**Review by mplorentz — CHANGES_REQUESTED** [review]
Functionally this works great! I left a few code comments though. Could find another way to link the tickets other than "Closes"? We don't want to close them right on merge to main anymore, since we are starting our UAT process.

**Comment by martindsq** [comment]
@mplorentz I didn't find a way to do what you request https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

Everything seems to close the issue.

Maybe it is easier if you adjust your workflow to this constraint? Maybe re-opening closed issues.

**Comment by mplorentz** [comment]
@martindsq I think the best way to do it is just to mention the ticket numbers in the description of the PR. This let's you easily view the ticket from the PR and the ticket will show a "author mentioned this" backlink to the PR. We won't get the nice links in the sidebar, but closing and reopening the ticket seems messier.

**Review by mplorentz — APPROVED** [review]

**Pull request merged**